### PR TITLE
Update to 2.0.0

### PR DESCRIPTION
**Pull request by Jrice1317**
python-lsp-black 2.0.0

**Destination channel:** {defaults}

### Links

- [Upstream repository](https://github.com/python-lsp/python-lsp-black/tree/v2.0.0)

### Explanation of changes:

- Update hash and version
- Only build for python versions 3.8, 3.9, 3.10 and 3.11 as listed [here](https://github.com/python-lsp/python-lsp-black/blob/v2.0.0/pyproject.toml#L2)
- Update dependencies
- Update pip check
- Add description

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -33,14 +33,18 @@ test:
   requires:
     - pip
   commands:
-    - pip check || true    # [not win]
+    - pip check || true    # [win]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
superflous, if pip check below is working

File: recipe/meta.yaml [hunk 2]
```diff
@@ -22,9 +22,9 @@ requirements:
     - wheel
   run:
     - python
-    - black >=22.3.0
+    - black >=23.11.0
     - python-lsp-server >=1.4.0
-    - toml
+    - toml  # [py<311]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - tomli  # [py<311]
```
https://github.com/python-lsp/python-lsp-black/blob/v2.0.0/setup.cfg#L24

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: d7eaeab2a377e96a82cc26afe2f8f2e1cf7c6eaefdcdeab026343e2e559dcce9
+  sha256: 8286d2d310c566844b3c116b824ada6fccfa6ba228b1a09a0526b74c04e0805f
 
 build:
   number: 0
-  skip: True  # [py<37]
-  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [py<38 or py>311]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  skip: True  # [py<38]
```

Comment (cbouss) [reply recipe/meta.yaml]: https://github.com/python-lsp/python-lsp-black/blob/v2.0.0/pyproject.toml#L2
This is just the target of their linting tool, but doesn't mean that this package doesn't work with 3.12.

**Review by danpetry — APPROVED** [review]
Approved based on @cbouss 's comments being addressed

File: recipe/meta.yaml (hunk 3 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: And they do test with 3.12, see https://github.com/python-lsp/python-lsp-black/blob/v2.0.0/.github/workflows/python.yml#L14.

Comment (Jrice1317) [reply recipe/meta.yaml]: Fair enough! Thanks

File: recipe/meta.yaml (hunk 2 above)
Comment (Jrice1317) [reply recipe/meta.yaml]: Ooh, good catch! Thanks

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**